src/content/mod_configuration.cpp

```diff
@@ -237,10 +238,24 @@ void ModConfiguration::resolveDependencies()
 				mod.unsatisfied_depends.insert(optdep);
 		}
 		// if a mod has no depends it is initially satisfied
-		if (mod.unsatisfied_depends.empty())
-			satisfied.push_back(mod);
-		else
-			unsatisfied.push_back(mod);
+		if (mod.unsatisfied_depends.empty()) {
+			if (mod.name==last_mod) {
+				auto it = satisfied.begin();
+			  satisfied.insert(it, mod);
+			} else {
+				if (mod.name!=first_mod) {
+					satisfied.push_back(mod);
+				} else {
+					m_sorted_mods.push_back(mod);
+				}
+			}
+		} else {
+      if ( (mod.name!=first_mod) && (mod.name!=last_mod) ) {
+			  unsatisfied.push_back(mod);
+      } else {
+        throw ModError("Mod selected as first/last by game is not allowed to have dependencies.");
+      }
```

Comment:
Space used as indentation here, should be tabs. Make sure to set this in your editor. Also helps to have whitespace visible

Reply:
Sorry. I am working on more projects, some of them using space, and some tabs indented. I change this setting often and sometimes I forget to do so. My fault.